spec/requests/clients/available_frames_controller_spec.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Clients::AvailableFramesController, type: :request do
+  let(:client) { create(:client) }
+  let(:client_id) { client.id }
+
+  before do
+    allow_any_instance_of(ActionDispatch::Request).to receive(:session).and_return({ user_id: client_id })
```

Comment:
うーん、allow_any_instance_ofはよっぽどの理由がない限り、あまり使わない方がいいですね。
理由は以下を参照
```
You can also configure the responses in the same manner.

This feature is sometimes useful when working with legacy code, though in general we
discourage its use for a number of reasons:

The rspec-mocks API is designed for individual object instances, but this feature operates on entire classes of objects. As a result there are some semantically confusing edge cases. For example, in expect_any_instance_of(Widget).to receive(:name).twice it isn't clear whether a specific instance is expected to receive name twice, or if two receives total are expected. (It's the former.)
Using this feature is often a design smell. It may be that your test is trying to do too much or that the object under test is too complex.
It is the most complicated feature of rspec-mocks, and has historically received the most bug reports. (None of the core team actively use it, which doesn't help.)
```
https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance

Reply:
↑ すばらしい心構え！

でもここって特定の request だけ stub するような書き方できますっけ？

Reply:
ここは使っているケースがログイン用のケースなので、許容範囲かなと思っている
他の方法でログイン状態を作ろうとしても結構手間が増えるし

Reply:
@shinpei-noda @yusuke-eto 自分だったら、```allow_any_instance_of(ActionDispatch::Request)```
を使わずに、ex_webの`include_context 'a user has signed in'` のように、shared_contextなどで、ユーザーがログインした状態を作るようにすると思うのですが、どちらが良いんでしょうか？

Reply:
shared_context側でもallow_any_instance_ofって使われてないんですか？

Reply:
ex_web の shared_context では実際に Devise (認証 Gem) の sign_in 使ってますね。

```rb
before { sign_in user }
```

今回のケースは allow_any_instance_of でいいと思いますけどね。

Reply:
@takaaki-kanetsuki @yusuke-eto @shinpei-noda

https://binarapps.com/blog/simple-authentication-with-bcrypt-and-warden/

wardenが使えれば行けそうな気はしますが，変更がデカイ気がするので，別でissue立ててやろうかなと思ってるんですがどうでしょう？

Reply:
時間もあまりないので，発展としてやることになりそうです